src/role_squadheal.js

```diff
@@ -54,65 +62,49 @@ roles.squadheal.preMove = function(creep, directions) {
       creep.memory.initialized = true;
     }
     if (squad.action === 'move') {
-      if (creep.room.name === squad.moveTarget) {
-        const nextExits = creep.room.find(creep.memory.route[creep.memory.routePos].exit);
-        const nextExit = nextExits[Math.floor(nextExits.length / 2)];
-        const range = creep.pos.getRangeTo(nextExit.x, nextExit.y);
-        if (range < 4) {
-          Memory.squads[creep.memory.squad].heal[creep.id].waiting = true;
-          //        if (Math.random() > 0.5 * (range - 2)) {
-          //          reverse = true;
-          //        }
-        }
+      if (creep.squadMove(squad, 4, false, 'heal')) {
+        return true;
       }
     }
   }
 };
 
 // TODO need to check if it works
 roles.squadheal.action = function(creep) {
-  if (creep.hits < creep.hitsMax) {
-    creep.log('action heal');
-    creep.heal(creep);
-    creep.say('exit');
-    const exit = creep.pos.findClosestByRange(FIND_EXIT);
-    creep.cancelOrder('move');
-    creep.cancelOrder('moveTo');
+  creep.heal(creep);
 
-    if (creep.pos.x === 0 || creep.pos.y === 0 || creep.pos.x === 49 || creep.pos.y === 49) {
-      return true;
-    }
-
-    const search = PathFinder.search(
-      creep.pos, {
-        pos: exit,
-        range: 0,
-      }, {
-        roomCallback: creep.room.getCostMatrixCallback(exit),
-        maxRooms: 1,
-      }
-    );
-
-    if (config.visualizer.enabled && config.visualizer.showPathSearches) {
-      visualizer.showSearch(search);
-    }
-
-    if (search.incomplete) {
-      creep.say('incomplete');
-      creep.log(creep.pos.getDirectionTo(exit.x, exit.y));
-      const returnCode = creep.move(creep.pos.getDirectionTo(exit.x, exit.y));
-      creep.log('rc: ' + returnCode);
-      return true;
+  if (creep.room.name !== creep.memory.routing.targetRoom) {
+    // creep.log('Not in room');
+    if (creep.hits < creep.hitsMax) {
+      creep.moveRandom();
+    } else {
+      // creep.log('delete?');
+      delete creep.memory.routing.reached;
     }
-    creep.move(creep.pos.getDirectionTo(search.path[0]));
-    delete creep.memory.routing.reached;
     return true;
+  } else {
+    creep.log('In room');
+    // TODO calculate if we would to flip directly back to the previous room
+    // get all towers and calculate their potential damage
+    // the damage is applied after the first tick
+    if (creep.hits < creep.hitsMax) {
+      creep.log('action heal');
+      creep.heal(creep);
+      creep.say('exit');
+      const exit = creep.pos.findClosestByRange(FIND_EXIT);
+      creep.moveTo(exit);
+    } else {
+      creep.log('mrandom');
+      creep.moveRandom();
+      creep.squadHeal();
+    }
   }
 
-  creep.squadHeal();
   return true;
 };
 
 roles.squadheal.execute = function(creep) {
-  //  creep.log('Execute!!!');
+  creep.log('Execute!!!');
+  creep.heal(creep);
+  creep.moveRandom();
```

Comment:
i think it's better to stop use execute no ? Maybe keep a log with debug config check for see if it is launch one day ^^
We should add execute calls to stats for see if we can remove them safely.

Reply:
That's why I enabled the log again.